src/Orleans/Configuration/GlobalConfiguration.cs

```diff
@@ -712,7 +802,61 @@ internal override void Load(XmlElement root)
+                            throw new FormatException("MultiClusterNetwork.GlobalServiceId cannot be blank");
+                        if (string.IsNullOrEmpty(ClusterId))
+                            throw new FormatException("MultiClusterNetwork.ClusterId cannot be blank");
+                        if (ClusterId.Contains(","))
+                            throw new FormatException("MultiClusterNetwork.ClusterId cannot contain commas: " + ClusterId);
+
+                        if (child.HasAttribute("DefaultMultiCluster"))
+                        {
+                            var toparse = child.GetAttribute("DefaultMultiCluster").Trim();
+                            if (string.IsNullOrEmpty(toparse))
+                            {
+                                DefaultMultiCluster = new List<string>(); // empty cluster
+                            }
+                            else
+                            {
+                                DefaultMultiCluster = toparse.Split(',').Select(id => id.Trim()).ToList();
+                                foreach (var id in DefaultMultiCluster)
+                                    if (string.IsNullOrEmpty(id))
+                                        throw new FormatException("MultiClusterNetwork.DefaultMultiCluster cannot contain blank cluster ids: " + toparse);
+                            }
+                        }
+                        if (child.HasAttribute("BackgroundGossipInterval"))
+                        {
+                            BackgroundGossipInterval = ConfigUtilities.ParseTimeSpan(child.GetAttribute("BackgroundGossipInterval"),
+                                "Invalid time value for the BackgroundGossipInterval attribute on the MultiClusterNetwork element");
+                        }
+                        if (child.HasAttribute("MaxMultiClusterGateways"))
+                        {
+                            MaxMultiClusterGateways = ConfigUtilities.ParseInt(child.GetAttribute("MaxMultiClusterGateways"),
+                                "Invalid time value for the MaxMultiClusterGateways attribute on the MultiClusterNetwork element");
+                        }
+                        var channels = new List<GossipChannelConfiguration>();
+                        foreach (XmlNode childchild in child.ChildNodes)
+                        {
+                            var channelspec = childchild as XmlElement;
+                            if (channelspec == null || channelspec.LocalName != "GossipChannel")
+                                continue;
+                            channels.Add(new GossipChannelConfiguration()
+                            {
+                                ChannelType = (GlobalConfiguration.GossipChannelType)
+                                   Enum.Parse(typeof(GlobalConfiguration.GossipChannelType), channelspec.GetAttribute("Type")),
+                                ConnectionString = channelspec.GetAttribute("ConnectionString")
```

Comment:
I think it is better to fail if ConnectionString is null, and force the user to specify it, than to use a default that seems to work superficially but does not really do the right thing